src/helpers.cpp

```diff
@@ -623,12 +623,12 @@ ada_really_inline void parse_prepared_path(std::string_view input,
     // but dots must as appear as '.', and they cannot be encoded because
     // the symbol '%' is not present.
     size_t previous_location = 0;  // We start at 0.
-    do {
-      size_t new_location = input.find('/', previous_location);
+    for (;;) {
```

Comment:
Why replace `do {} while (true)` with `for(;;;) {}`

How is that better?

Reply:
You'd have to scroll 50/500/50000000000000000000000000 lines of code just to figure out that it's an infinite loop, then you have to scroll 50/500/50000000000000000000000000 lines of code, to go back where you started.